api/app/src/main/kotlin/packit/controllers/PacketController.kt

```diff
@@ -77,4 +81,29 @@ class PacketController(
             .headers(response.second)
             .body(response.first)
     }
+
+    @GetMapping("/{name}/{id}/zip")
+    @PreAuthorize("@authz.canReadPacket(#root, #id, #name)")
+    // TODO: Verify authorization for accessing specific files: whether user has access to all files, or artefacts only.
+    fun downloadZip(
+        @PathVariable id: String,
+        @PathVariable name: String,
+        @RequestParam(required = true) hashes: List<String>,
+        @RequestParam(required = true) filenames: List<String>,
```

Comment:
These are hashes and filenames which should be in sync e.g. hashes[0] is hash for filenames[0]? Would be better to send that in a more structured way to avoid length mismatch e.g. a map of filenames to hashes. 

Do you actually need the filename, or could that be discoverable from the hash? (Maybe not - I suppose in theory you could have two files with different names and same content/hash in a packet..?)

Need to return 400 if invalid hash for packet (or filename not matching packet metadata..?)
(..or {id} not in {name})

Reply:
> Do you actually need the filename, or could that be discoverable from the hash?

I believe we could look up the filename from the packet's metadata using the hash, but this would imply one extra request to retrieve the necessary metadata. I'd rather complicate the request param (as a map of hashes and filenames) than add another request in.